lib/index.js

```diff
@@ -101,3 +109,50 @@ exports.defaultDocumentLoader = defaultDocumentLoader;
 
 // expose for testing purposes
 exports._caches = require('./caches');
+
+/**
+ * Takes in a request to a kms system from a proxy and validates it.
+ *
+ * @param {object} options - Options to use.
+ * @param {Map<string, Array<string>>} options.allowedHosts - A Map
+ *   with a host and an optional Array of allowed ips.
+ * @param {string} options.host - An URL's host.
+ * @param {string} options.ip - An ip for the request.
+ *
+ * @throws {BedrockError} - A NotAllowedError if ip rules are broken.
+ *
+ * @returns {object} An object with the verify result and any errors.
+ */
+exports.validateRequest = ({allowedHosts, host, ip}) => {
```

Comment:
I expect that this will need to be `async` looking forward to when a database lookup is involved.

Reply:
I think we should actually just get rid of all the host checks and check IPs only at this point. The ability to create keystores with a certain host name will be handled via zcaps elsewhere. This check is really about ensuring there's some network-based authz going on.